src/advanced/components/ProductSelect.tsx

```diff
@@ -0,0 +1,64 @@
+import type { Product } from '../stores/products';
+import { formatPrice } from '../utils/price';
+import { isSoldOut } from '../utils/quantity';
+import { getSaleRecord, hasNoneSale } from '../utils/sale-event';
+
+interface Props {
+  products: Product[];
+  productId: string;
+  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+}
+
+export function ProductSelect({ products, productId, onChange }: Props) {
+  return (
+    <select
+      value={productId}
+      onChange={onChange}
+      className="w-full p-3 border border-gray-300 rounded-lg text-base mb-3"
+    >
+      {products.map((product) => {
+        const { textContent, ...props } = getOptionProps(product);
+
+        return (
+          <option key={product.id} {...props}>
+            {textContent}
+          </option>
+        );
+      })}
```

Comment:
textContent 를 childen 으로 바꿔주면 이런 것도 될 것 같아요

```suggestion
      {products.map((product) => <options {...getOptionProps(product)} />}
```

Reply:
와우 깔끔쓰..

Reply:
> textContent 를 childen 으로 바꿔주면 이런 것도 될 것 같아요

생각을 못 해봤네요 👍